modules/cluster-logging-collector-rsyslog.adoc

```diff
@@ -0,0 +1,185 @@
+// Module included in the following assemblies:
+//
+// * logging/cluster-logging-external.adoc
+
+[id="cluster-logging-collector-rsyslog_{context}"]
+= Sending logs using the Fluentd *syslog* plug-in (RFC 3164)
+
+You can use the Fluentd *syslog* plug-in to send a copy of your cluster Fluentd logs to an external syslog server, 
+and not the default Elasticsearch. Note the following about the *syslog* plug-in:
+
+* uses syslog protocol (RFC 3164), not RFC 5424;
+* does not support TLS and thus, is not secure;
+* does not provide Kubernetes metadata, systemd data, or other metadata.
```

Comment:
The RFC 3164 protocol is very limited in terms of the metadata you can provide to the remote server - all of the metadata is ingested, and is in the fluentd pipeline available for other outputs, just not syslog RFC 3164

Reply:
> The RFC 3164 protocol is very limited in terms of the metadata you can provide to the remote server - all of the metadata is ingested, and is in the fluentd pipeline available for other outputs, just not syslog RFC 3164
@richm This means that meta will be "dropped" by syslog then?

Reply:
> This means that meta will be "dropped" by syslog then?

yes

Reply:
hmm - maybe not - If I'm reading https://github.com/openshift/origin-aggregated-logging/blob/master/fluentd/out_syslog_buffered.rb#L162 and https://github.com/openshift/origin-aggregated-logging/blob/master/fluentd/out_syslog.rb#L138 correctly, if you have
```
use_record true
```
in the plugin config, it should send at least some of the kubernetes metadata like this:
```
namespace_name=namespace_name, container_name=container_name, pod_name=pod_name, message=message
```
however, it doesn't seem to be working.  I'm trying this code locally in ruby and I can't get content to have a value:
```ruby
record = {"message" => "message","kubernetes" => {"namespace_name" => "namespace_name","container_name" => "container_name","pod_name" => "pod_name"}}
payload_key = "message"
use_record = true
puts "record #{record} use_record #{use_record}"
content = if use_record && (record.key?('kubernetes')) && record[payload_key]
  puts "in if true"
  tmp = (((record['kubernetes']).key?('namespace_name')) ? 'namespace_name=' + record['kubernetes']['namespace_name'] + ', ' : '' ) + \
        (((record['kubernetes']).key?('container_name')) ? 'container_name=' + record['kubernetes']['container_name'] + ', ' : '' ) + \
        (((record['kubernetes']).key?('pod_name')) ? 'pod_name=' + record['kubernetes']['pod_name'] + ', ' : '' ) + \
        payload_key + '=' + record[payload_key]
  puts "tmp #{tmp}"
  tmp
else
  puts "in if false"
  record[payload_key]
end
puts "content = "#{content}"
```
`tmp` inside the `if` clause prints the correct information, but `content` is always empty - this seems like some really obscure ruby stuff?  @nhosoi any insight here?

Reply:
This new behavior of `use_record` was added by https://github.com/openshift/origin-aggregated-logging/pull/955/files so it has been in the product since 3.9 or so.